From cfe6fd2243a629a5a0085524fc320b9d31d16dfa Mon Sep 17 00:00:00 2001
From: Adrien Ricciardi <aricciardi@baylibre.com>
Date: Tue, 3 Oct 2023 10:59:41 +0200
Subject: [PATCH 17/38] goil: template: spider: Improved the CAN library
 definition to avoid issues with the 'spider_can.h' file.

Sometimes the 'tpl_can_controller_t' type was not found in the 'spider_can.h'
file.

By looking at the preprocessed file, it seemed that the 'spider_can.h' file
was included too soon and the files it was including (more precisely 'Can.h')
content was not present.

Signed-off-by: Adrien Ricciardi <aricciardi@baylibre.com>
---
 goil/templates/config/cortex-a/armv8/spider/config.oil | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/goil/templates/config/cortex-a/armv8/spider/config.oil b/goil/templates/config/cortex-a/armv8/spider/config.oil
index 206c42ed..9368994e 100755
--- a/goil/templates/config/cortex-a/armv8/spider/config.oil
+++ b/goil/templates/config/cortex-a/armv8/spider/config.oil
@@ -137,11 +137,15 @@ CPU spider {
     CFILE = "../netif/bridgeif_fdb.c";
   };
 
+  LIBRARY can_driver {
+    PATH = "cortex-a/armv8/spider/drivers/can";
+    CFILE = "spider_can.c";
+  };
+
   LIBRARY can {
+    NEEDS = can_driver;
     PATH = "../net/can";
     CFILE = "tpl_can_core.c";
-    CFILE = "../../machines/cortex-a/armv8/spider/drivers/can/spider_can.c";
-    CHEADER = "../../machines/cortex-a/armv8/spider/drivers/can/spider_can.h";
   };
 
   POSTBUILD all {
-- 
2.34.1

